### DISCUSSION TITLE
Começar por Front ou Back para ser FullStack?

### DISCUSSION THREAD

**Discussion by jhongalante** [frontendbr/forum#1259] [Discussão]
Boa tarde a todos.

Para alguém que está começando na área de desenvolvimento web/mobile, vcs acham que vale mais a pena começar pelo front ou backend?

**Comment by wmartins** [comment]
Oi @JhonGalante, eu acho bem complicado te responder coisas simplesmente como "frontend" ou "backend". Acho que qualquer resposta que alguém vá te dar, sem entender o teu contexto pessoal, dificilmente vai ser assertiva (e construtiva).

Se tu quiser, dá mais alguns detalhes aqui, como por exemplo a tua experiência, teus interesses, o porquê de tu estar trilhando esse caminho e etc.

Para não te deixar com uma resposta somente filosófica, tenta entender o que significa ser um desenvolvedor full stack e quais as expectativas de alguém nesse cargo/papel. Isso é tão importante quanto a parte tecnológica em si. 

Com isso, vê qual dessas áreas tu tem mais interesse (que te "brilha mais os olhos") e vai fundo. Particularmente, eu funciono melhor quando estou motivado com alguma coisa. Com isso, não faria sentido estudar uma área que não me traz "alegria" de estudar.

Enfim, boa sorte!

**Comment by alexandreramosdev** [comment]
Ai man!!  Eu começei a estudar este ano, e estou começando a estudar pelo front-end ! mas no meu dos estudos sempre tem um ou outro conceito de back-end que tenho que apreende !   Hoje eu maior foco é no JS pois vai bem tanto no front e no back-end

**Comment by evertonthepaula** [comment]
Cara, eu vou assumir que você é realmente iniciante e vou te dar uma dica que gostaria que tivessem me dado. Aprenda Clean Code, Solid, Paradigmas de programação e design patterns comuns. Escolha uma ou várias linguagens que vão te ajudar a aprender tudo isso, só então você vai para front ou back.

**Comment by victorferreira** [comment]
TL;DR: front

Vou tentar elencar alguns pontos que justificam meu pensamento.

1. Você só precisa do browser pra começar. Esqueça ter que configurar um servidor Linux, instalar Apache, MySQL e PHP. O navegador tem tudo o que você precisa pra começar.
2. É muito fácil ver o resultado do que você faz. Seja com um console.log ou renderizar uma tag no DOM. Nada de usar Qt pra construir interfaces ou tentar fazer malabarismos com o terminal.
3. A Web na suas mãos. Ferramentas como o [Glitch](https://glitch.com) permitem que você faça algo legal em pouco tem se preocupar como fazer deploy numa CDN, por exemplo. E hoje o que não faltam são ferramentas para facilitar a sua vida.
4. Javascript. Veja a lei de Atwood: "any application that can be written in JavaScript, will eventually be written in JavaScript".

Como iniciante é muito importante desenvolver um ritmo constante de desenvolvimento, e é muito fácil ficar frustrado com um monte de barreiras desnecessárias pra quem está apenas começando. No mais, não pare!

**Comment by didiraja** [comment]
Pra mim Front é a porta de entrada pra tudo [inclusive drogas], tanto desenvolvimento web quando programação em geral, pela simplicidade no código.

**Comment by allansduarte** [comment]
@JhonGalante acredito que a melhor maneira para se tornar um dev fullstack é aliar o conhecimento a projetos reais, ou seja, aos projetos que você trabalha atualmente. Veja bem, um dev fullstack é basicamente um engenheiro que pode lidar com todo o trabalho de bancos de dados, servidores, engenharia de sistemas e clients. Dependendo da demanda que vier para você, que pode ser mobile stack, web stack ou uma aplicação nativa você estuda e une a sua demanda, que se torna uma nova experiência. No caso de ser backend ou frontend sugiro exercitar as duas tecnologias ao mesmo tempo o que é naturalmente possível, pois seu objetivo não é ser especialista, mas sim um generalista (fullstack) que pode lidar com várias tecnologias. E não há mal algum em ser generalista.
